lib/opener-registry.js

```diff
@@ -0,0 +1,56 @@
+/** @babel */
+
+import _ from 'lodash'
+import fs from 'fs-plus'
+import path from 'path'
+
+export default class OpenerRegistry {
+  constructor () {
+    this.createOpeners()
+  }
+
+  async createOpeners () {
+    const moduleDir = path.join(__dirname, 'openers')
+    const entries = fs.readdirSync(moduleDir)
+    this.openers = entries.map(entry => {
+      const OpenerImpl = require(path.join(moduleDir, entry))
+      return new OpenerImpl()
+    })
+    // Sort the openers into decreasing priority
+    this.openers = _.orderBy(this.openers, [opener => opener.hasSynctex(), opener => opener.canOpenInBackground()], ['desc', 'desc'])
```

Comment:
Extract this "capability list" into a separate variable for readability? Call it `capabilityPrioritization` or something along those lines. Maybe simply `capabilities`?